pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -423,6 +499,89 @@ pub mod pallet {
 
 			T::InboundQueue::submit(domain_address, incoming_msg)
 		}
+
+		/// Convenience method for manually processing an outbound message.
+		///
+		/// If the execution fails, the message gets moved to the
+		/// `FailedOutboundMessages` storage.
+		#[pallet::weight(T::WeightInfo::process_outbound_message())]
+		#[pallet::call_index(6)]
+		pub fn process_outbound_message(
+			origin: OriginFor<T>,
+			nonce: T::OutboundMessageNonce,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let (domain, sender, message) =
+				OutboundMessageQueue::<T>::get(nonce).ok_or(Error::<T>::OutboundMessageNotFound)?;
+
+			OutboundMessageQueue::<T>::remove(nonce);
+
+			match Self::process_message(domain.clone(), sender.clone(), message.clone()) {
+				Ok(_) => {
+					Self::deposit_event(Event::<T>::OutboundMessageExecutionSuccess {
+						nonce,
+						domain,
+						sender,
+						message,
+					});
+
+					Ok(())
+				}
+				Err(e) => {
+					Self::deposit_event(Event::<T>::OutboundMessageExecutionFailure {
+						nonce,
+						domain: domain.clone(),
+						sender: sender.clone(),
+						message: message.clone(),
+						error: e.error,
+					});
+
+					FailedOutboundMessages::<T>::insert(nonce, (domain, sender, message, e.error));
+
+					Err(e.error)
+				}
+			}
+		}
+
+		/// Manually process a failed outbound message.
+		#[pallet::weight(T::WeightInfo::process_failed_outbound_message())]
+		#[pallet::call_index(7)]
+		pub fn process_failed_outbound_message(
+			origin: OriginFor<T>,
+			nonce: T::OutboundMessageNonce,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let (domain, sender, message, _) = FailedOutboundMessages::<T>::get(nonce)
+				.ok_or(Error::<T>::OutboundMessageNotFound)?;
+
+			match Self::process_message(domain.clone(), sender.clone(), message.clone()) {
+				Ok(_) => {
+					Self::deposit_event(Event::<T>::OutboundMessageExecutionSuccess {
+						nonce,
+						domain,
+						sender,
+						message,
+					});
+
+					FailedOutboundMessages::<T>::remove(nonce);
```

Comment:
Q: AFAICS, fault messages which fail to succeed will congest our `FailedOutboundMessages` storage because they cannot be removed. Is that desired for now to keep things simple?

cc @mustermeiszer

Reply:
We could also add some extrinsics for removing these as long as there's an admin doing it. Please let us know what you think @mustermeiszer